### ISSUE TITLE
Unable to decommission a node

### ISSUE THREAD

**Issue by shantanugithub** [scylladb/scylladb#14446]
Getting below error while decommissioning a node from the cluster. These are the logs from the instance I'm trying to decommission --

```
Jun 29 15:11:21 ip-172-31-xx-239 scylla[103942]:  [shard 0] storage_service - decommission[ecb39391-ee2a-453e-a7a8-b9449908ad4f]: Operation failed, sync_nodes={172.31.xx.xx, 172.31.xx.xx, 172.31.xx.xx, 172.31.xx.239}: std::invalid_argument (The configuration must have at least one voter)
Jun 29 15:11:21 ip-172-31-xx-239 scylla[103942]:  [shard 0] storage_service - decommission[ecb39391-ee2a-453e-a7a8-b9449908ad4f]: Stopped heartbeat_updater
Jun 29 15:11:21 ip-172-31-xx-239 scylla[103942]:  [shard 0] storage_service - decommission[ecb39391-ee2a-453e-a7a8-b9449908ad4f]: Started decommission_abort[ecb39391-ee2a-453e-a7a8-b9449908ad4f]: ignore_nodes=[], leaving_nodes=[172.31.xx.239], replace_nodes={}, bootstrap_nodes={}, repair_tables=[]
```

Also attaching the zip of node logs since beginning for debugging.

*Installation details*
Scylla version : 5.2.2-0.20230521.9dd70a58c3f9
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI): AWS AMI
Cluster details - 3 nodes of i4i.large type and 1 node of i3en.6xlarge

[logs_node_1_decom.log.gz](https://github.com/scylladb/scylladb/files/11904975/logs_node_1_decom.log.gz)
[logs_node_2.log.gz](https://github.com/scylladb/scylladb/files/11904977/logs_node_2.log.gz)
[logs_node_3.log.gz](https://github.com/scylladb/scylladb/files/11904978/logs_node_3.log.gz)
[logs_node_4.log.gz](https://github.com/scylladb/scylladb/files/11904979/logs_node_4.log.gz)

**Comment by kbr-scylla** [comment]
According to the logs, you have bootstrapped each node as a separate cluster:
172-31-xx-90:
```
    Jun 07 16:30:22 ip-172-31-xx-90 scylla[3653]:  [shard 0] init - seeds={172.31.xx.90}, listen_address=172.31.xx.90, broadcast_address=172.31.xx.90

    Jun 07 16:30:23 ip-172-31-xx-90 scylla[3653]:  [shard 0] raft_group0 - server 96d7d756-d289-40d9-906a-6765bb097205 found no local group 0. Discovering...
    Jun 07 16:30:23 ip-172-31-xx-90 scylla[3653]:  [shard 0] raft_group0 - server 96d7d756-d289-40d9-906a-6765bb097205 found group 0 with group id 807bad80-0522-11ee-9881-24d66eb0700d, leader 96d7d756-d289-40d9-906a-6765bb097205
    Jun 07 16:30:23 ip-172-31-xx-90 scylla[3653]:  [shard 0] raft_group0 - Server 96d7d756-d289-40d9-906a-6765bb097205 chosen as discovery leader; bootstrapping group 0 from scratch

    Jun 07 16:39:09 ip-172-31-xx-90 scylla[4480]:  [shard 0] init - seeds={172.31.xx.90}, listen_address=172.31.xx.90, broadcast_address=172.31.xx.90

    Jun 07 16:39:10 ip-172-31-xx-90 scylla[4480]:  [shard 0] raft_group0 - server e82cb54b-01a2-40cb-90ae-776e92d4eab4 found no local group 0. Discovering...
    Jun 07 16:39:10 ip-172-31-xx-90 scylla[4480]:  [shard 0] raft_group0 - server e82cb54b-01a2-40cb-90ae-776e92d4eab4 found group 0 with group id ba741cb0-0523-11ee-8400-4f2e77edb35e, leader e82cb54b-01a2-40cb-90ae-776e92d4eab4
    Jun 07 16:39:10 ip-172-31-xx-90 scylla[4480]:  [shard 0] raft_group0 - Server e82cb54b-01a2-40cb-90ae-776e92d4eab4 chosen as discovery leader; bootstrapping group 0 from scratch
```

172-31-xx-83:
```
    Jun 07 16:30:23 ip-172-31-xx-83 scylla[3665]:  [shard 0] init - seeds={172.31.xx.83}, listen_address=172.31.xx.83, broadcast_address=172.31.xx.83

    Jun 07 16:30:25 ip-172-31-xx-83 scylla[3665]:  [shard 0] raft_group0 - setup_group0: joining group 0...
    Jun 07 16:30:25 ip-172-31-xx-83 scylla[3665]:  [shard 0] raft_group0 - server 3e3be8c4-7c24-4ae2-92be-047268cb943d found no local group 0. Discovering...
    Jun 07 16:30:25 ip-172-31-xx-83 scylla[3665]:  [shard 0] raft_group0 - server 3e3be8c4-7c24-4ae2-92be-047268cb943d found group 0 with group id 814e18b0-0522-11ee-9466-3cdc32fb5d0c, leader 3e3be8c4-7c24-4ae2-92be-047268cb943d
    Jun 07 16:30:25 ip-172-31-xx-83 scylla[3665]:  [shard 0] raft_group0 - Server 3e3be8c4-7c24-4ae2-92be-047268cb943d chosen as discovery leader; bootstrapping group 0 from scratch

    Jun 07 16:36:40 ip-172-31-xx-83 scylla[4417]:  [shard 0] init - seeds={172.31.xx.83}, listen_address=172.31.xx.83, broadcast_address=172.31.xx.83

    Jun 07 16:39:12 ip-172-31-xx-83 scylla[4494]:  [shard 0] raft_group0 - setup_group0: joining group 0...
    Jun 07 16:39:12 ip-172-31-xx-83 scylla[4494]:  [shard 0] raft_group0 - server 7c745c49-5bbf-4ad5-af8a-cfdce3e0f2b5 found no local group 0. Discovering...
    Jun 07 16:39:12 ip-172-31-xx-83 scylla[4494]:  [shard 0] raft_group0 - server 7c745c49-5bbf-4ad5-af8a-cfdce3e0f2b5 found group 0 with group id bbd448f0-0523-11ee-8b2f-843271c93029, leader 7c745c49-5bbf-4ad5-af8a-cfdce3e0f2b5
    Jun 07 16:39:12 ip-172-31-xx-83 scylla[4494]:  [shard 0] raft_group0 - Server 7c745c49-5bbf-4ad5-af8a-cfdce3e0f2b5 chosen as discovery leader; bootstrapping group 0 from scratch
```

172-31-xx-215:
```
    Jun 27 19:35:48 ip-172-31-7-215 scylla[3971]:  [shard  0] init - seeds={172.31.7.215}, listen_address=172.31.7.215, broadcast_address=172.31.7.215

    Jun 27 19:35:49 ip-172-31-7-215 scylla[3971]:  [shard  0] raft_group0 - server 4e88258a-0e51-41a1-bb55-c7cbbd4b791a found no local group 0. Discovering...
    Jun 27 19:35:49 ip-172-31-7-215 scylla[3971]:  [shard  0] raft_group0 - server 4e88258a-0e51-41a1-bb55-c7cbbd4b791a found group 0 with group id b8122770-14f3-11ee-810d-44bc007b69e3, leader 4e88258a-0e51-41a1-bb55-c7cbbd4b791a
    Jun 27 19:35:49 ip-172-31-7-215 scylla[3971]:  [shard  0] raft_group0 - Server 4e88258a-0e51-41a1-bb55-c7cbbd4b791a chosen as discovery leader; bootstrapping group 0 from scratch

    Jun 27 19:56:23 ip-172-31-7-215 scylla[4932]:  [shard  0] init - seeds={172.31.xx.239, 172.31.xx.83, 172.31.xx.90}, listen_address=172.31.7.215, broadcast_address=172.31.7.215

    Jun 27 19:56:24 ip-172-31-7-215 scylla[4932]:  [shard  0] raft_group0 - setup_group0: joining group 0...
    Jun 27 19:56:24 ip-172-31-7-215 scylla[4932]:  [shard  0] raft_group0 - server 18d71165-8e50-407a-a53a-61188d0d19fb found no local group 0. Discovering...
    Jun 27 19:56:24 ip-172-31-7-215 scylla[4932]:  [shard  0] raft_group0 - server 18d71165-8e50-407a-a53a-61188d0d19fb found group 0 with group id ba741cb0-0523-11ee-8400-4f2e77edb35e, leader e82cb54b-01a2-40cb-90ae-776e92d4eab4
    Jun 27 19:56:24 ip-172-31-7-215 scylla[4932]:  [shard  0] raft_group0 - Server 18d71165-8e50-407a-a53a-61188d0d19fb is starting group 0 with id ba741cb0-0523-11ee-8400-4f2e77edb35e
```

172-31-xx-239:
```
    Jun 07 16:30:24 ip-172-31-xx-239 scylla[3667]:  [shard 0] init - seeds={172.31.xx.239}, listen_address=172.31.xx.239, broadcast_address=172.31.xx.239

    Jun 07 16:30:25 ip-172-31-xx-239 scylla[3667]:  [shard 0] raft_group0 - setup_group0: joining group 0...
    Jun 07 16:30:25 ip-172-31-xx-239 scylla[3667]:  [shard 0] compaction - [Compact system.local 81911520-0522-11ee-b90c-c04f6b816b91] Compacted 2 sstables to [/var/lib/scylla/data/system/local-7ad54392bcdd35a684174e047860b377/mc-6-big-Data.db:level=0]. 28k
    B to 14kB (~50% of original) in 12ms = 1MB/s. ~256 total partitions merged to 1.
    Jun 07 16:30:25 ip-172-31-xx-239 scylla[3667]:  [shard 0] raft_group0 - server d1659153-9886-4002-bb03-8712c8b03ee8 found no local group 0. Discovering...
    Jun 07 16:30:25 ip-172-31-xx-239 scylla[3667]:  [shard 0] raft_group0 - server d1659153-9886-4002-bb03-8712c8b03ee8 found group 0 with group id 81988f30-0522-11ee-b90c-c04f6b816b91, leader d1659153-9886-4002-bb03-8712c8b03ee8
    Jun 07 16:30:25 ip-172-31-xx-239 scylla[3667]:  [shard 0] raft_group0 - Server d1659153-9886-4002-bb03-8712c8b03ee8 chosen as discovery leader; bootstrapping group 0 from scratch


    Jun 07 16:36:37 ip-172-31-xx-239 scylla[4417]:  [shard 0] init - seeds={172.31.xx.239}, listen_address=172.31.xx.239, broadcast_address=172.31.xx.239

    Jun 07 16:39:08 ip-172-31-xx-239 scylla[4493]:  [shard 0] raft_group0 - server 6277418b-13e1-4682-b8fa-5509016821cf found no local group 0. Discovering...
    Jun 07 16:39:08 ip-172-31-xx-239 scylla[4493]:  [shard 0] raft_group0 - server 6277418b-13e1-4682-b8fa-5509016821cf found group 0 with group id b92b7010-0523-11ee-bef7-3a6602a3cbb1, leader 6277418b-13e1-4682-b8fa-5509016821cf
    Jun 07 16:39:08 ip-172-31-xx-239 scylla[4493]:  [shard 0] raft_group0 - Server 6277418b-13e1-4682-b8fa-5509016821cf chosen as discovery leader; bootstrapping group 0 from scratch
```

That is: when you bootstrapped `172-31-xx-90`, you set `seeds={172-31-xx-90}`. This created a single-node cluster consisting of only `172-31-xx-90`.

Then you bootstrapped `172-31-xx-83` with `seeds={172-31-xx-83}`. Which created a completely different single-node cluster consisting of only `172-31-xx-83`.

And so on, all 4 nodes were bootstrapped this way.

At the end, you modified the `seeds` configuration on all nodes which caused the different clusters to gossip with each other and connect (which Scylla should prevent by detecting `group-id` mismatch - I'll file a bug). But still, from Raft point of view they are different clusters.

Please consult the documentation for the correct way to bootstrap a cluster:
https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/cluster-management/create-cluster-multidc.html

In particular what you misconfigured was the `seeds` parameter.

The correct way is as follows:
1. the first node points to itself, e.g. if your first node is `172-31-xx-90`, then you set `seeds={172-31-xx-90}` and you start it first.
2. the second node points to the first node, e.g. if your second node is `172-31-xx-83` then you set `seeds={172-31-xx-90}`
3. the third node points to the first OR the second node OR both (your choice).

And so on; the Nth node you bootstrap must point to one of the existing nodes in seeds.

The documentation says this:
> seeds - Specify the IP of the first node and **only the first node**. New nodes will use the IP of this seed node to connect to the cluster and learn the cluster topology and state.

That is indeed the simplest way to bootstrap a cluster.
Remember to boot the nodes sequentially:
>  After you have installed and configured Scylla and edit scylla.yaml on all nodes, using one node as a seed. Start the seed node, and once it is in UN state, repeat for all the other nodes, each after the previous is in UN state.

**Comment by kbr-scylla** [comment]
Now, as what to do with this cluster - do you store any data in it? If not, it would be simplest to just throw it away completely and bootstrap a fresh cluster.

If you do have data and you need to keep the cluster, we'll have to think of something else.

**Comment by kbr-scylla** [comment]
> According to the logs, you have bootstrapped each node as a separate cluster:

To be precise - I see that `172-31-xx-215` was actually bootstrapped correctly (the first time it started it used itself as seed, but then you removed data directories and configured seeds correctly to use `{172.31.xx.239, 172.31.xx.83, 172.31.xx.90}`). That's why you have a two-node cluster in there, and two single-node clusters.

**Comment by kbr-scylla** [comment]
cc @kostja @gleb-cloudius

**Comment by kostja** [comment]
A separate issue has been filed to improve user experience when there is an operator error following our installation instructions (#14448). https://opensource.docs.scylladb.com/stable/operating-scylla/procedures/cluster-management/create-cluster.

**Comment by mykaul** [comment]
I feel this is solved with topology over raft and 6.0? @kbr-scylla ?

**Comment by kbr-scylla** [comment]
It was solved in 5.4 in fact, with group 0 ID equality check in gossiper.
